enhancements/testing/ephemeral-agent-installer-e2e-tests.md

```diff
@@ -0,0 +1,275 @@
+they predate the ephemeral agent based installer project. There also isn't an easy way
+to use an existing routine to just launch a generic ISO image which would have been
+something we could have re-purposed. 
+
+### API Extensions
+
+Not applicable at this time.
+
+### Implementation Details/Notes/Constraints [optional]
+
+What are the caveats to the implementation? What are some important details that
+didn't come across above. Go in to as much detail as necessary here. This might
+be a good place to talk about core concepts and how they relate.
+
+### Risks and Mitigations
+
+The risks for this proposal are highlighted in the motivation section. The risks we 
+have are:
+
+1) Creating a third e2e framework for installers that isn't reusable. Meaning in
+the future we will have this same issue if a new installer project comes online.
+
+2) The third e2e framework becomes very complex and is hard to understand where the 
+failures are coming from.
+
+As for how we mitigate these issues I believe that we can create a modular e2e framework
+that will mitigate this problem for future installer projects. To mitigate the second risk
+we need to make sure that each test has a clear failure message. We also need to encapsulate
+red herring messages that happen during set up and installation of components and mark these
+accordingly.
+
+## Design Details
+
+* The e2e framework will be a collection of Ansible playbooks and roles
+* Using roles to define reusable steps scoped to units of work
+* Use playbooks to set up common e2e workflows and chains for the project
+
+Thinking about a general pattern of how to organize the repository:
+
+```
+playbooks/
+  example-e2e-workflow.yml
+roles/
+    common/
+        tasks/
+        handlers/
+        library/
+        files/
+        templates/
+        vars/
+        defaults/
+        meta/
+    example_role/
+        tasks/
+        defaults/
+        meta/
+```
+
+This is pretty open ended and meant to be. I just believe that focusing on making
+reusable roles here for the steps is key.
+
+These are my thoughts about Openshift CI and the step registry and making it easy to understand. 
+Copy the `pre`, `test`, and `post` pattern locally in the playbooks and include the playbooks like so:
+
+```
+# File structure:
+playbooks/
+  example-e2e-workflow.yml
+  example-e2e-workflow_pre.yml
+  example-e2e-workflow_test.yml
+  example-e2e-workflow_post.yml
+
+# example-e2e-workflow.yml
+- hosts: localhost
+
+- name: Run e2e-workflow Pre steps
+  import_playbook: example-e2e-workflow_pre.yml
+
+- name: Run e2e-workflow Test steps
+  import_playbook: example-e2e-workflow_test.yml
+
+- name: Run e2e-workflow Post steps
+  import_playbook: example-e2e-workflow_test.yml
+```
+
+This is just a general idea and would like to keep this pretty open ended at the moment.
+The first priority will be to run these playbooks locally but how to understand this in the CI
+is being taken into consideration.
```

Comment:
It would be useful to have a little more detail about what the tools need to do, to get an idea of how that might be broken up into roles and how much, if any, of what already exists would be useful to reuse.